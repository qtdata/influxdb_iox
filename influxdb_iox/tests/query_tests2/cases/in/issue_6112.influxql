-- Query tests derived from https://github.com/influxdata/influxdb_iox/issues/6112
-- IOX_SETUP: InfluxQLSelectSupport

--
-- Single measurement queries
--

-- Validates expected data is returned
-- Projection wildcard, all tags and fields
-- IOX_COMPARE: sorted
SELECT * FROM m0;

-- No matching measurement
SELECT * FROM non_existent;

-- Projection wildcard, only tags
-- IOX_COMPARE: sorted
SELECT *::tag, f64 FROM m0;

-- Projection wildcard, only fields
-- IOX_COMPARE: sorted
SELECT *::field FROM m0;

-- Projection regex, mixture of tags and fields
-- IOX_COMPARE: sorted
SELECT /64|tag0/ FROM m0;

-- Projection specific tags and fields
-- IOX_COMPARE: sorted
SELECT f64, tag0 FROM m0;

-- Explicitly select time column
-- IOX_COMPARE: sorted
SELECT f64, tag0, time FROM m0;

-- arithmetic operators
-- IOX_COMPARE: sorted
SELECT f64, f64 * 2, i64, i64 + i64 FROM m0;

-- bitwise operators
-- IOX_COMPARE: sorted
SELECT i64, i64 & 1 FROM m0;

-- Automatic type coercion integer â†’ float
-- IOX_COMPARE: sorted
SELECT f64 + i64 FROM m0;

-- Type cast postfix operator
SELECT f64, f64::integer FROM m0;

-- Column alias behaviour
-- IOX_COMPARE: sorted
SELECT f64 AS f64_2, f64, f64, f64 FROM m0 LIMIT 1;

--
-- WHERE clause
--

-- Single tag
-- IOX_COMPARE: sorted
SELECT tag0, f64 FROM m0 WHERE tag0 = 'val00';

-- IOX_COMPARE: sorted
SELECT tag0, f64 FROM m0 WHERE tag0 =~ /^val0(1|2)/;

-- Conjunction (AND)
-- IOX_COMPARE: sorted
SELECT /tag(0|1)/, f64 FROM m0 WHERE tag0 = 'val00' AND tag1 = 'val10';

-- Disjunction (OR)
-- IOX_COMPARE: sorted
SELECT /tag(0|1)/, f64 FROM m0 WHERE tag0 = 'val00' OR tag1 = 'val10';

-- arithmetic
-- IOX_COMPARE: sorted
SELECT f64 FROM m0 WHERE f64 > 10 + 10;

-- bitwise
-- IOX_COMPARE: sorted
SELECT i64 FROM m0 WHERE i64 & 1 = 0;

-- time bounds

-- timestamp format %Y-%M-%D
-- IOX_COMPARE: sorted
SELECT i64 FROM m0 WHERE time > '2022-10-31';

-- timestamp format %Y-%M-%D %h:%m:%s
-- IOX_COMPARE: sorted
SELECT i64 FROM m0 WHERE time > '2022-10-31 02:00:10';

-- now() and duration
-- NOTE: 100000d is > 270 years, so this test should be ok for a while.
--       However, if this test is still in use in 270 years and it starts failing,
--       try increasing the number of days ðŸ˜‚
-- IOX_COMPARE: sorted
SELECT i64 FROM m0 WHERE time > now() - 100000d;

-- NOT NULL test
-- WHERE tag1 != '' is the equivalent to tag1 IS NOT NULL
-- TODO(sgc): This is working, but likely by accident
-- IOX_COMPARE: sorted
SELECT tag1, f64 FROM m0 WHERE tag1 != '';

-- NULL test
-- WHERE tag1 = '' is the equivalent to tag1 IS NULL
-- TODO(sgc): Not working, as expected
-- -- IOX_COMPARE: sorted
-- SELECT tag1, f64 FROM m0 WHERE tag1 = '';

--
-- LIMIT and OFFSET clauses
-- NOTE: these are working, but due to incorrect default ordering
-- some tests fail
--

SELECT tag0, f64 FROM m0 LIMIT 1;

SELECT tag0, f64 FROM m0 WHERE tag0 = 'val00' LIMIT 2 OFFSET 1;

SELECT tag0, f64 FROM m0 LIMIT 1 OFFSET 1;

-- OFFSET clause, no LIMIT clause
-- TODO(sgc): Fails due to a bug in InfluxQL that utilises the following optimisation
--  https://github.com/influxdata/influxdb/blob/dee8977d2c6598cb2d17e9334ea997c99853640a/tsdb/engine/tsm1/iterator.gen.go#L344-L347
--  which breaks after returning the first point after the offset, because itr.opt.Limit == 0
-- SELECT tag0, f64 FROM m0 OFFSET 1;

--
-- Sort ordering
--

-- No GROUP BY clause

-- Default sort: expected output should default to ORDER BY TIME asc
SELECT * FROM m0;

-- Sort time in descending order
SELECT * FROM m0 ORDER BY time DESC;

--
-- Scalar functions in projection
--

-- Validate all scalar functions with a float field
SELECT
    f64,
    abs(f64 * -1),
    sin(f64),
    cos(f64),
    tan(f64),
    asin(1/f64),
    acos(1/f64),
    atan(f64),
    atan2(f64, 2),
    exp(f64),
-- TODO(sgc): Dependent on https://github.com/apache/arrow-datafusion/issues/5206
--    log(f64, 8),
    ln(f64),
    log2(f64),
    log10(f64),
    sqrt(f64),
    pow(f64, 2),
    floor(f64),
    ceil(f64),
    round(f64)
FROM m0 LIMIT 1;

-- Validate all scalar functions with an integer field
SELECT
    i64,
    abs(i64 * -1),
    sin(i64),
    cos(i64),
    tan(i64),
-- TODO(sgc): Not coerced to float, so returns incorrect result
--    asin(1/i64),
    acos(1/i64),
    atan(i64),
    atan2(i64, 2),
    exp(i64),
-- TODO(sgc): Dependent on https://github.com/apache/arrow-datafusion/issues/5206
--    log(i64, 8),
    ln(i64),
    log2(i64),
    log10(i64),
    sqrt(i64),
    pow(i64, 2),
    floor(i64),
    ceil(i64),
    round(i64)
FROM m0 LIMIT 1;

-- Deviation from InfluxQL is that NaNs are not coalesced to NULL
-- The InfluxQL compatibility later will be responsible for this translation
SELECT f64, asin(f64), acos(f64) FROM m0 LIMIT 1;

-- INF support
SELECT f64, pow(f64, pow(2, 10)) FROM m0 LIMIT 1;

--
-- TZ clause support
--

-- Interpret date/time (%Y-%M-%D %h:%m:%s) in timezone specified by TZ clause
-- TODO(sgc): condition is correct, but `time` column is not display in local timezone
--  as DataFusion does not support timestamp with timezone data types, and displaying
--  the values in the local timezone

---- SELECT f64 FROM m0 WHERE time = '2022-10-31 13:00:00' TZ('Australia/Hobart');
---- SELECT f64 FROM m0 WHERE time = '2022-10-31T13:00:00Z' TZ('Australia/Hobart');

--
-- Validate conditional expressions
--

SELECT f64 FROM m0 WHERE f64 >= 19 + 0.5;
SELECT f64 FROM m0 WHERE f64 - 0.5 >= 19;
SELECT f64 FROM m0 WHERE str = 'h' + 'i';

-- should be odd floating point numbers
SELECT f64 FROM m0 WHERE f64::integer & 1 = 1;

-- should be even floating point numbers
SELECT f64 FROM m0 WHERE f64::integer & 1 = 0;

-- These two deviate from InfluxQL, as the casting is optional
SELECT f64 FROM m0 WHERE f64 & 1 = 1;
SELECT f64 FROM m0 WHERE f64 & 1 = 0;

-- Expressions with incompatible types should succeed and may or may not return results
SELECT f64 FROM m0 WHERE f64 >= 19 + 0.5 OR str = 1;
SELECT f64 FROM m0 WHERE f64 >= 19.5 AND str = 1;

-- Missing columns should succeed and may or may not return results
SELECT f64 FROM m0 WHERE f64 >= 19 + 0.5 OR non_existent = 1;
SELECT f64 FROM m0 WHERE f64 >= 19.5 AND non_existent = 1;

-- Regular expressions should succeed and may or may not return results
SELECT f64 FROM m0 WHERE f64 >= 19.5 AND f64 =~ /foo/;
SELECT f64 FROM m0 WHERE f64 >= 19.5 OR f64 =~ /foo/;

--
-- Validate column expressions
--

-- float types
SELECT tag0, f64, f64 * 0.5, f64 + str FROM m0 WHERE f64 > 19;
-- string types
SELECT tag0, str, str + 'foo', str * 5 FROM m0 WHERE f64 > 19;
-- integer types
SELECT tag0, i64, i64 * 0.5, i64 + f64::integer, i64 & 1 FROM m0 WHERE f64 > 19;

-- non-existing column
SELECT f64, non_existing, f64 + non_existing FROM m0 WHERE f64 > 19;

--
-- Multiple measurements in the FROM clause
--

SELECT usage_idle, bytes_used FROM cpu, disk;