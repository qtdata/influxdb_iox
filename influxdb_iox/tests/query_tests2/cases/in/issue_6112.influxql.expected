-- Test Setup: InfluxQLSelectSupport
-- InfluxQL: SELECT * FROM m0;
-- Results After Sorting
+----------------------+------+-----+-----+-------+-------+
| time                 | f64  | i64 | str | tag0  | tag1  |
+----------------------+------+-----+-----+-------+-------+
| 2022-10-31T02:00:00Z | 10.1 | 101 | hi  | val00 |       |
| 2022-10-31T02:00:00Z | 10.4 | 101 | lo  | val02 |       |
| 2022-10-31T02:00:00Z | 11.3 | 211 | lo  | val01 |       |
| 2022-10-31T02:00:10Z | 18.9 | 211 | lo  | val00 | val10 |
| 2022-10-31T02:00:10Z | 21.2 | 211 | hi  | val00 |       |
| 2022-10-31T02:00:20Z | 11.2 | 191 | lo  | val00 |       |
| 2022-10-31T02:00:30Z | 19.2 | 392 | lo  | val00 |       |
+----------------------+------+-----+-----+-------+-------+
-- InfluxQL: SELECT *::tag, f64 FROM m0;
-- Results After Sorting
+----------------------+-------+-------+------+
| time                 | tag0  | tag1  | f64  |
+----------------------+-------+-------+------+
| 2022-10-31T02:00:00Z | val00 |       | 10.1 |
| 2022-10-31T02:00:00Z | val01 |       | 11.3 |
| 2022-10-31T02:00:00Z | val02 |       | 10.4 |
| 2022-10-31T02:00:10Z | val00 | val10 | 18.9 |
| 2022-10-31T02:00:10Z | val00 |       | 21.2 |
| 2022-10-31T02:00:20Z | val00 |       | 11.2 |
| 2022-10-31T02:00:30Z | val00 |       | 19.2 |
+----------------------+-------+-------+------+
-- InfluxQL: SELECT *::field FROM m0;
-- Results After Sorting
+----------------------+------+-----+-----+
| time                 | f64  | i64 | str |
+----------------------+------+-----+-----+
| 2022-10-31T02:00:00Z | 10.1 | 101 | hi  |
| 2022-10-31T02:00:00Z | 10.4 | 101 | lo  |
| 2022-10-31T02:00:00Z | 11.3 | 211 | lo  |
| 2022-10-31T02:00:10Z | 18.9 | 211 | lo  |
| 2022-10-31T02:00:10Z | 21.2 | 211 | hi  |
| 2022-10-31T02:00:20Z | 11.2 | 191 | lo  |
| 2022-10-31T02:00:30Z | 19.2 | 392 | lo  |
+----------------------+------+-----+-----+
-- InfluxQL: SELECT /64|tag0/ FROM m0;
-- Results After Sorting
+----------------------+------+-----+-------+
| time                 | f64  | i64 | tag0  |
+----------------------+------+-----+-------+
| 2022-10-31T02:00:00Z | 10.1 | 101 | val00 |
| 2022-10-31T02:00:00Z | 10.4 | 101 | val02 |
| 2022-10-31T02:00:00Z | 11.3 | 211 | val01 |
| 2022-10-31T02:00:10Z | 18.9 | 211 | val00 |
| 2022-10-31T02:00:10Z | 21.2 | 211 | val00 |
| 2022-10-31T02:00:20Z | 11.2 | 191 | val00 |
| 2022-10-31T02:00:30Z | 19.2 | 392 | val00 |
+----------------------+------+-----+-------+
-- InfluxQL: SELECT f64, tag0 FROM m0;
-- Results After Sorting
+----------------------+------+-------+
| time                 | f64  | tag0  |
+----------------------+------+-------+
| 2022-10-31T02:00:00Z | 10.1 | val00 |
| 2022-10-31T02:00:00Z | 10.4 | val02 |
| 2022-10-31T02:00:00Z | 11.3 | val01 |
| 2022-10-31T02:00:10Z | 18.9 | val00 |
| 2022-10-31T02:00:10Z | 21.2 | val00 |
| 2022-10-31T02:00:20Z | 11.2 | val00 |
| 2022-10-31T02:00:30Z | 19.2 | val00 |
+----------------------+------+-------+
-- InfluxQL: SELECT f64, tag0, time FROM m0;
-- Results After Sorting
+------+-------+----------------------+
| f64  | tag0  | time                 |
+------+-------+----------------------+
| 10.1 | val00 | 2022-10-31T02:00:00Z |
| 10.4 | val02 | 2022-10-31T02:00:00Z |
| 11.2 | val00 | 2022-10-31T02:00:20Z |
| 11.3 | val01 | 2022-10-31T02:00:00Z |
| 18.9 | val00 | 2022-10-31T02:00:10Z |
| 19.2 | val00 | 2022-10-31T02:00:30Z |
| 21.2 | val00 | 2022-10-31T02:00:10Z |
+------+-------+----------------------+
-- InfluxQL: SELECT f64, f64 * 2, i64, i64 + i64 FROM m0;
-- Results After Sorting
+----------------------+------+-------+-----+---------+
| time                 | f64  | f64_1 | i64 | i64_i64 |
+----------------------+------+-------+-----+---------+
| 2022-10-31T02:00:00Z | 10.1 | 20.2  | 101 | 202     |
| 2022-10-31T02:00:00Z | 10.4 | 20.8  | 101 | 202     |
| 2022-10-31T02:00:00Z | 11.3 | 22.6  | 211 | 422     |
| 2022-10-31T02:00:10Z | 18.9 | 37.8  | 211 | 422     |
| 2022-10-31T02:00:10Z | 21.2 | 42.4  | 211 | 422     |
| 2022-10-31T02:00:20Z | 11.2 | 22.4  | 191 | 382     |
| 2022-10-31T02:00:30Z | 19.2 | 38.4  | 392 | 784     |
+----------------------+------+-------+-----+---------+
-- InfluxQL: SELECT i64, i64 & 1 FROM m0;
-- Results After Sorting
+----------------------+-----+-------+
| time                 | i64 | i64_1 |
+----------------------+-----+-------+
| 2022-10-31T02:00:00Z | 101 | 1     |
| 2022-10-31T02:00:00Z | 101 | 1     |
| 2022-10-31T02:00:00Z | 211 | 1     |
| 2022-10-31T02:00:10Z | 211 | 1     |
| 2022-10-31T02:00:10Z | 211 | 1     |
| 2022-10-31T02:00:20Z | 191 | 1     |
| 2022-10-31T02:00:30Z | 392 | 0     |
+----------------------+-----+-------+
-- InfluxQL: SELECT f64 + i64 FROM m0;
-- Results After Sorting
+----------------------+---------+
| time                 | f64_i64 |
+----------------------+---------+
| 2022-10-31T02:00:00Z | 111.1   |
| 2022-10-31T02:00:00Z | 111.4   |
| 2022-10-31T02:00:00Z | 222.3   |
| 2022-10-31T02:00:10Z | 229.9   |
| 2022-10-31T02:00:10Z | 232.2   |
| 2022-10-31T02:00:20Z | 202.2   |
| 2022-10-31T02:00:30Z | 411.2   |
+----------------------+---------+
-- InfluxQL: SELECT f64 AS f64_2, f64, f64, f64 FROM m0 LIMIT 1;
-- Results After Sorting
+----------------------+-------+------+-------+-------+
| time                 | f64_2 | f64  | f64_1 | f64_3 |
+----------------------+-------+------+-------+-------+
| 2022-10-31T02:00:00Z | 10.1  | 10.1 | 10.1  | 10.1  |
+----------------------+-------+------+-------+-------+
-- InfluxQL: SELECT tag0, f64 FROM m0 WHERE tag0 = 'val00';
-- Results After Sorting
+----------------------+-------+------+
| time                 | tag0  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:00Z | val00 | 10.1 |
| 2022-10-31T02:00:10Z | val00 | 18.9 |
| 2022-10-31T02:00:10Z | val00 | 21.2 |
| 2022-10-31T02:00:20Z | val00 | 11.2 |
| 2022-10-31T02:00:30Z | val00 | 19.2 |
+----------------------+-------+------+
-- InfluxQL: SELECT tag0, f64 FROM m0 WHERE tag0 =~ /^val0(1|2)/;
-- Results After Sorting
+----------------------+-------+------+
| time                 | tag0  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:00Z | val01 | 11.3 |
| 2022-10-31T02:00:00Z | val02 | 10.4 |
+----------------------+-------+------+
-- InfluxQL: SELECT /tag(0|1)/, f64 FROM m0 WHERE tag0 = 'val00' AND tag1 = 'val10';
-- Results After Sorting
+----------------------+-------+-------+------+
| time                 | tag0  | tag1  | f64  |
+----------------------+-------+-------+------+
| 2022-10-31T02:00:10Z | val00 | val10 | 18.9 |
+----------------------+-------+-------+------+
-- InfluxQL: SELECT /tag(0|1)/, f64 FROM m0 WHERE tag0 = 'val00' OR tag1 = 'val10';
-- Results After Sorting
+----------------------+-------+-------+------+
| time                 | tag0  | tag1  | f64  |
+----------------------+-------+-------+------+
| 2022-10-31T02:00:00Z | val00 |       | 10.1 |
| 2022-10-31T02:00:10Z | val00 | val10 | 18.9 |
| 2022-10-31T02:00:10Z | val00 |       | 21.2 |
| 2022-10-31T02:00:20Z | val00 |       | 11.2 |
| 2022-10-31T02:00:30Z | val00 |       | 19.2 |
+----------------------+-------+-------+------+
-- InfluxQL: SELECT f64 FROM m0 WHERE f64 > 10 + 10;
-- Results After Sorting
+----------------------+------+
| time                 | f64  |
+----------------------+------+
| 2022-10-31T02:00:10Z | 21.2 |
+----------------------+------+
-- InfluxQL: SELECT i64 FROM m0 WHERE i64 & 1 = 0;
-- Results After Sorting
+----------------------+-----+
| time                 | i64 |
+----------------------+-----+
| 2022-10-31T02:00:30Z | 392 |
+----------------------+-----+
-- InfluxQL: SELECT i64 FROM m0 WHERE time > '2022-10-31';
-- Results After Sorting
+----------------------+-----+
| time                 | i64 |
+----------------------+-----+
| 2022-10-31T02:00:00Z | 101 |
| 2022-10-31T02:00:00Z | 101 |
| 2022-10-31T02:00:00Z | 211 |
| 2022-10-31T02:00:10Z | 211 |
| 2022-10-31T02:00:10Z | 211 |
| 2022-10-31T02:00:20Z | 191 |
| 2022-10-31T02:00:30Z | 392 |
+----------------------+-----+
-- InfluxQL: SELECT i64 FROM m0 WHERE time > '2022-10-31 02:00:10';
-- Results After Sorting
+----------------------+-----+
| time                 | i64 |
+----------------------+-----+
| 2022-10-31T02:00:20Z | 191 |
| 2022-10-31T02:00:30Z | 392 |
+----------------------+-----+
-- InfluxQL: SELECT i64 FROM m0 WHERE time > now() - 100000d;
-- Results After Sorting
+----------------------+-----+
| time                 | i64 |
+----------------------+-----+
| 2022-10-31T02:00:00Z | 101 |
| 2022-10-31T02:00:00Z | 101 |
| 2022-10-31T02:00:00Z | 211 |
| 2022-10-31T02:00:10Z | 211 |
| 2022-10-31T02:00:10Z | 211 |
| 2022-10-31T02:00:20Z | 191 |
| 2022-10-31T02:00:30Z | 392 |
+----------------------+-----+
-- InfluxQL: SELECT tag1, f64 FROM m0 WHERE tag1 != '';
-- Results After Sorting
+----------------------+-------+------+
| time                 | tag1  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:10Z | val10 | 18.9 |
+----------------------+-------+------+
-- InfluxQL: SELECT tag0, f64 FROM m0 LIMIT 1;
+----------------------+-------+------+
| time                 | tag0  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:00Z | val00 | 10.1 |
+----------------------+-------+------+
-- InfluxQL: SELECT tag0, f64 FROM m0 WHERE tag0 = 'val00' LIMIT 2 OFFSET 1;
+----------------------+-------+------+
| time                 | tag0  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:10Z | val00 | 21.2 |
| 2022-10-31T02:00:10Z | val00 | 18.9 |
+----------------------+-------+------+
-- InfluxQL: SELECT tag0, f64 FROM m0 LIMIT 1 OFFSET 1;
+----------------------+-------+------+
| time                 | tag0  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:00Z | val01 | 11.3 |
+----------------------+-------+------+
-- InfluxQL: SELECT * FROM m0;
+----------------------+------+-----+-----+-------+-------+
| time                 | f64  | i64 | str | tag0  | tag1  |
+----------------------+------+-----+-----+-------+-------+
| 2022-10-31T02:00:00Z | 10.1 | 101 | hi  | val00 |       |
| 2022-10-31T02:00:00Z | 11.3 | 211 | lo  | val01 |       |
| 2022-10-31T02:00:00Z | 10.4 | 101 | lo  | val02 |       |
| 2022-10-31T02:00:10Z | 21.2 | 211 | hi  | val00 |       |
| 2022-10-31T02:00:10Z | 18.9 | 211 | lo  | val00 | val10 |
| 2022-10-31T02:00:20Z | 11.2 | 191 | lo  | val00 |       |
| 2022-10-31T02:00:30Z | 19.2 | 392 | lo  | val00 |       |
+----------------------+------+-----+-----+-------+-------+
-- InfluxQL: SELECT * FROM m0 ORDER BY time DESC;
+----------------------+------+-----+-----+-------+-------+
| time                 | f64  | i64 | str | tag0  | tag1  |
+----------------------+------+-----+-----+-------+-------+
| 2022-10-31T02:00:30Z | 19.2 | 392 | lo  | val00 |       |
| 2022-10-31T02:00:20Z | 11.2 | 191 | lo  | val00 |       |
| 2022-10-31T02:00:10Z | 21.2 | 211 | hi  | val00 |       |
| 2022-10-31T02:00:10Z | 18.9 | 211 | lo  | val00 | val10 |
| 2022-10-31T02:00:00Z | 10.1 | 101 | hi  | val00 |       |
| 2022-10-31T02:00:00Z | 11.3 | 211 | lo  | val01 |       |
| 2022-10-31T02:00:00Z | 10.4 | 101 | lo  | val02 |       |
+----------------------+------+-----+-----+-------+-------+
-- InfluxQL: SELECT f64, abs(f64 * -1), sin(f64), cos(f64), tan(f64), asin(1/f64), acos(1/f64), atan(f64), atan2(f64, 2), exp(f64), ln(f64), log2(f64), log10(f64), sqrt(f64), pow(f64, 2), floor(f64), ceil(f64), round(f64) FROM m0 LIMIT 1;
+----------------------+------+------+---------------------+---------------------+--------------------+--------------------+-------------------+------------------+--------------------+-------------------+-------------------+--------------------+--------------------+--------------------+--------------------+-------+------+-------+
| time                 | f64  | abs  | sin                 | cos                 | tan                | asin               | acos              | atan             | atan2              | exp               | ln                | log2               | log10              | sqrt               | pow                | floor | ceil | round |
+----------------------+------+------+---------------------+---------------------+--------------------+--------------------+-------------------+------------------+--------------------+-------------------+-------------------+--------------------+--------------------+--------------------+--------------------+-------+------+-------+
| 2022-10-31T02:00:00Z | 10.1 | 10.1 | -0.6250706488928821 | -0.7805681801691837 | 0.8007893029375109 | 0.0991723838059207 | 1.471623942988976 | 1.47210806614649 | 1.3753055265462157 | 24343.00942440838 | 2.312535423847214 | 3.3362833878644325 | 1.0043213737826426 | 3.1780497164141406 | 102.00999999999999 | 10    | 11   | 10    |
+----------------------+------+------+---------------------+---------------------+--------------------+--------------------+-------------------+------------------+--------------------+-------------------+-------------------+--------------------+--------------------+--------------------+--------------------+-------+------+-------+
-- InfluxQL: SELECT i64, abs(i64 * -1), sin(i64), cos(i64), tan(i64), acos(1/i64), atan(i64), atan2(i64, 2), exp(i64), ln(i64), log2(i64), log10(i64), sqrt(i64), pow(i64, 2), floor(i64), ceil(i64), round(i64) FROM m0 LIMIT 1;
+----------------------+-----+-----+---------------------+--------------------+--------------------+--------------------+-------------------+-----------+----------------------------------------------+------------------+-------------------+--------------------+-------------------+-------+-------+------+-------+
| time                 | i64 | abs | sin                 | cos                | tan                | acos               | atan              | atan2     | exp                                          | ln               | log2              | log10              | sqrt              | pow   | floor | ceil | round |
+----------------------+-----+-----+---------------------+--------------------+--------------------+--------------------+-------------------+-----------+----------------------------------------------+------------------+-------------------+--------------------+-------------------+-------+-------+------+-------+
| 2022-10-31T02:00:00Z | 101 | 101 | 0.45202578717835057 | 0.8920048697881602 | 0.5067526002248183 | 1.5707963267948966 | 1.560895660206908 | 1.5509969 | 73070599793680670000000000000000000000000000 | 4.61512051684126 | 6.658211482751795 | 2.0043213737826426 | 10.04987562112089 | 10201 | 101   | 101  | 101   |
+----------------------+-----+-----+---------------------+--------------------+--------------------+--------------------+-------------------+-----------+----------------------------------------------+------------------+-------------------+--------------------+-------------------+-------+-------+------+-------+
-- InfluxQL: SELECT f64, asin(f64), acos(f64) FROM m0 LIMIT 1;
+----------------------+------+------+------+
| time                 | f64  | asin | acos |
+----------------------+------+------+------+
| 2022-10-31T02:00:00Z | 10.1 | NaN  | NaN  |
+----------------------+------+------+------+
-- InfluxQL: SELECT f64, pow(f64, pow(2, 10)) FROM m0 LIMIT 1;
+----------------------+------+-----+
| time                 | f64  | pow |
+----------------------+------+-----+
| 2022-10-31T02:00:00Z | 10.1 | inf |
+----------------------+------+-----+